aten/src/ATen/native/native_functions.yaml

```diff
@@ -2941,6 +2941,10 @@
 
 - func: max_pool1d(Tensor self, int[1] kernel_size, int[1] stride=[], int[1] padding=0, int[1] dilation=1, bool ceil_mode=False) -> Tensor
 
+- func: _max_pool1d_forward(Tensor self, int[1] kernel_size, int[1] stride=[], int[1] padding=0, int[1] dilation=1, bool ceil_mode=False) -> Tensor
+  dispatch:
+    CompositeExplicitAutograd: _max_pool1d_forward
+
```

Comment:
@kshitij12345 do you have a description of what the problem was? We want to try to avoid adding new operators as much as we can

Reply:
It's these lines : https://github.com/pytorch/pytorch/blob/e1b84e1b6bd688c7b3bff3e7742c1f8875966e6e/aten/src/ATen/native/MaxPooling.cpp#L76-L79

There is a proper kernel for forward with CPU.